security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,131 @@
+--- Makefile.orig	2021-09-09 19:58:05.000000000 +1000
++++ Makefile	2023-06-12 13:16:16.000000000 +1000
+@@ -23,10 +23,10 @@
+ ID := $(NAME)-$(VERSION)
+ 
+ DESTDIR :=
+-PREFIX := $(DESTDIR)/usr
++PREFIX := $(DESTDIR)/opt/local
```

Comment:
Thanks. How do I do that? I assumed patch files were static.

Reply:
I haven't tried this myself but it should be done in a "post-patch" block. At that point the should exist. 

Typically, you would add a place holder (e.g., "@@MACPORTS@@") in the patchfile and then do a reinplace of that.

But I agree with @herbygillot, you should just change the Makefile such that it will take values from the environment if set; in that case MacPorts will likely set most of things to their correct values, potentially with needing the "makefile" PortGroup.

Reply:
Thanks. I found example uses of reinplace and it's now working.

Reply:
Apologies in advance for the things you already know, but this is public so others might read it:

Makefile macros can always be overridden via the command line, rather than via the environment. Overriding via the environment only works if there is no value in the Makefile itself. For some macros, like CFLAGS, it can make sense to set them via the environment (only), because they might not strictly be necessary, so they can be left empty in the Makefile (although I would have thought that most Makefiles for C programs would set CFLAGS in the Makefile unless they are anticipating something like the macports legacysupport environment). But for some macros, like PREFIX, some non-empty value needs to be in the Makefile, so the best way to override those is via the command line (or a configure script that changes the Makefile). The make install command in macports overrides DESTDIR via the command line (that might be due to the makefile port group, not sure). It could/might also do it for PREFIX, but I haven't seen it do that (possibly because the prefix was set by configure (in a different port with use_configure yes)).

I've change the Portfile to include the makefile port group, because there is a Makefile, and I've changed the Makefile so that the PREFIX no longer includes DESTDIR (that was wrong). Now they can be overridden independently of each other. So, if the PREFIX were overridden on the make install command line, it should be fine without the need for reinplace. But it would probably require overriding the make install command so as to include PREFIX=${prefix}. However, the reinplace changes more than just the Makefile (i.e., the config file and sshdo itself). So it's probably best to keep using it for now. The next version of sshdo will have a configure script that will be able to do everything that the reinplace is doing now for v1.1.

Reply:
Yes, MacPorts base overrides a Makefile's DESTDIR at destroot time via the `destroot.destdir` variable.

MacPorts base doesn't do anything with a PREFIX variable. The makefile portgroup does set the PREFIX environment variable. It does not by default override such a variable. The Makefile can be written such that that is not necessary, for example:

```
PREFIX ?= /usr/local
```

If the Makefile has not been written that way, then the makefile portgroup can be asked to override the PREFIX environment variable by using:

```
makefile.override-append PREFIX
```

Reply:
Thanks. The new-since-writing-the-above (v1.1.1) configure script sets the prefix in the Makefile to ${prefix}. The Portfile makes that happen with `configure.args --macports-prefix=${prefix}`. It also changes the prefix in some other files that need to match.